agent.go

```diff
@@ -635,3 +646,37 @@ VERIFYCONSULSERVER:
 	a.logger.Debug("Consul agent and all servers are running compatible versions with ESM")
 	return nil
 }
+
+// PartitionOrEmpty returns the partition if it exists, otherwise returns an empty string.
+func (a *Agent) PartitionOrEmpty() string {
+	if a.config == nil || a.config.Partition == "" {
+		return ""
+	}
+	return a.config.Partition
+}
+
+// HasPartition checks if the partition is valid and calls the callback with the partition if it has any.
+func (a *Agent) HasPartition(callback func(partition string)) {
```

Comment:
this reads a little strange to me, is this checking if a partition is valid then it should be returning a `bool` to indicate if it's valid (and should probably be named `ValidPartition`) though this isn't checking if the partition is valid, it's just checking if the partition on the agent is non-emtpy and non-default, I'd probably get rid of this function entirely and re-write where it's used to just assign the output of `PartitionOrEmtpy`